### PR TITLE
added comments for included libraries

### PR DESCRIPTION
**Pull request by shiv-tyagi**
Added few comments for the libraries included in the file.

### CODE REVIEW

**Comment by shiv-tyagi** [comment]
Also rewrote a comment in 'Attitude.cpp' which earlier was confusing and less understandable.

**Comment by hendjoshsr71** [comment]
Hi, thanks for your first contribution! 

You will definitely be asked to rename the commit to something like, "ArduCopter: add comments for included libraries"
And you may be asked to squash it to just one commit. See [https://ardupilot.org/dev/docs/git-interactive-rebase.html#git-interactive-rebase](url)

A single commit is always prefaced by the library being changed and only makes changes to that library.

File: ArduCopter/Copter.h [hunk 1]
```diff
@@ -62,11 +62,11 @@
 #include <AC_InputManager/AC_InputManager_Heli.h>   // Heli specific pilot input handling library
 #include <AP_Arming/AP_Arming.h>
 #include <AP_SmartRTL/AP_SmartRTL.h>
```
Comment (hendjoshsr71) [inline ArduCopter/Copter.h:64]: Why not add a comment to these as well as `AP_Parachute.h` and `AP_Proximity.h` below and finish it up :) ?
```suggestion
#include <AP_Arming/AP_Arming.h>
#include <AP_SmartRTL/AP_SmartRTL.h>
```

File: ArduCopter/Copter.h [hunk 2]
```diff
@@ -28,18 +28,18 @@
 #include <AP_HAL/AP_HAL.h>
 
 // Common dependencies
-#include <AP_Common/AP_Common.h>
-#include <AP_Common/Location.h>
-#include <AP_Param/AP_Param.h>
-#include <StorageManager/StorageManager.h>
+#include <AP_Common/AP_Common.h>        //Common definitions and utility routines for the ArduPilot libraries.
```
Comment (hendjoshsr71) [inline ArduCopter/Copter.h:31]: Nitpicks.
Probably should add a space between all of these `// `. 
Also you won't see many periods unless it is a full sentence
```suggestion
#include <AP_Common/AP_Common.h>        // Common definitions and utility routines for the ArduPilot libraries
```
Usually, all of the comments get lined up together which it looks like you did with most of your changes.  :)
Pre-existing of course, but if you got adventurous you could line up the rest.

**Comment by shiv-tyagi** [comment]
Hi @hendjoshsr71 , Thanks for helping me out with this. Being a complete beginner to open source and ArduPilot, I am not quite aware of few things. But I'll make sure to keep such things in mind in future. While trying to squash the commits into one, I got few conflicts and I tried to resolve them. I am not sure if I did it correctly. Can you please tell me if I have done it correctly? Also, if I need to do something to fix the things, please let me know. Thanks in advance.

**Review by khancyr — CHANGES_REQUESTED** [review]
we don't use merge commit.
Try to follow instructions here https://ardupilot.org/dev/docs/git-interactive-rebase.html# to clean up your PR.
You should end up with Only one commit with all comments changes

**Comment by shiv-tyagi** [comment]
I'll rework on it and create a fresh PR again keeping all these points in mind. Thanks for helping me out with this @khancyr.

**Comment by hendjoshsr71** [comment]
@shiv-tyagi It is not necessary to close the request. Once you have removed all of the merge commits and got it down to one commit you can `force-push` to your github branch to over write commits.

I'm not sure what tool you are using for git gui or command line? But in SourceTree there is an option to enable force-pushing. 

I'm guessing what you did regarding the merge commits is that you tried to update your working branch with the latest changes from Ardupilot:master. If you want to do that, what you need to do is pull changes from ArduPilot:master into
shiv-tyagi:master. Then rebase your branch for this PR onto master.

This process is described here: https://ardupilot.org/dev/docs/git-rebase.html

**Comment by hendjoshsr71** [comment]
> that is just comments, not sure it is worth the trouble

Ture enough, but it is where Copter's life starts and I believe on the Wiki we direct people to this section?

**Pull request closed without merging**